src/adapter/etl-adapter-filesystem/src/Flow/ETL/Adapter/Filesystem/AwsS3Stream.php

```diff
@@ -42,11 +42,7 @@ protected function filesystem() : Filesystem
              */
             $clientOptions = \array_merge(
                 [
-                    'credentials' => [
-                        'key' => '',
-                        'secret' => '',
-                    ],
-                    'region' => '',
+                    'region' => 'us-east-1',
```

Comment:
I'm honestly not sure about default region, I think I would rather prefer to throw an exception when the region is not set. Is there anything that makes us-east-1 special enough to make it default?

Reply:
It's just because it's the default region AWS applies.
IMHO I'd rather avoid merging any default configuration :)

Reply:
> IMHO I'd rather avoid merging any default configuration :)

Same, can't say out of my head why it was implemented this way, I need to take another look later in the codebase but as far as I remember there was a reason. I will look into this later and get back to you

Reply:
I was playing with it a bit and I think we can skip any defaults, when region or credentials are missing, S3 client will throw an exception which I think is good since it's S3 responsibility, not flow. 
Could you remove those merges from both, S3 and Azure clients and in the meantime I'm gonna prepare some examples for the docs that would explain how to use them both?

Reply:
You got it. I'll take care of it tomorrow

Reply:
sounds good, examples should be available any moment: https://github.com/flow-php/flow/pull/1065

Reply:
@jguittard would you mind if I would take it over from here and adjust your PR so it can be merged?